src/main/java/org/testng/TestNG.java

```diff
@@ -875,9 +875,7 @@ private void initializeCommandLineSuitesGroups() {
     }
   }
   private void addReporter(Class<? extends IReporter> r) {
-    if (! m_reporters.contains(r)) {
-      m_reporters.add(ClassHelper.newInstance(r));
-    }
+    m_reporters.add(ClassHelper.newInstance(r));
```

Comment:
You mean `Set<IReporter>`. But we can imagine was the intention here and I think it was not to remove the check ;)